app/controllers/concerns/project_concern.rb

```diff
@@ -0,0 +1,13 @@
+module ProjectConcern
+  extend ActiveSupport::Concern
+
+  private
+
+  def change_date(project)
```

Comment:
@linesarefuzzy You are correct I think. Good thinking. My mistake. Fortunately I think there was not much wasted work. We still needed to abstract, just to a different place. So all the `@urlComponent`s can I think just become `projects`.